redaxo/src/addons/backup/pages/import.upload.php

```diff
@@ -16,16 +16,15 @@
 
 @set_time_limit(0);
 
-if ('' != $impname) {
-    $impname = basename($impname);
-
-    if ('dbimport' == $function && '.sql' != substr($impname, -4, 4)) {
-        $impname = '';
-    } elseif ('fileimport' == $function && '.tar.gz' != substr($impname, -7, 7)) {
-        $impname = '';
-    } elseif (('delete' == $function || 'download' == $function) && '.sql' != substr($impname, -4, 4) && '.tar.gz' != substr($impname, -7, 7)) {
-        $impname = '';
-    }
+/** @psalm-taint-escape text */
```

Comment:
@staabm - mind putting up a PR to psalm to move the `text` sinks into a `file` one? A generic one for read/write/delete is fine. - I can submit the documentation if that convinces you ;-)

Example on how to do so: https://github.com/vimeo/psalm/pull/4604 

Reason: This will break once I change it myself otherwise ;-)
Ref https://github.com/vimeo/psalm/issues/4596

Reply:
@lukasreschke I had a quick look around the codebase regarding your suggestion.

I am not sure, I could deliver a solution right now, with the knowledge I have of the psalm codebase

Reply:
@staabm  np :-) - It's just a bit of boilerplate shifting around, but it could indeed be significantly better documented :)

Did so at https://github.com/vimeo/psalm/pull/4630